common/src/primitives/id/mod.rs

```diff
@@ -66,26 +69,32 @@ impl<'de> serde::Deserialize<'de> for H256 {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Encode, Decode, serde::Serialize, serde::Deserialize)]
+#[derive(PartialEq, Eq, Encode, Decode, serde::Serialize, serde::Deserialize)]
 #[serde(transparent)]
-pub struct Id<T: ?Sized> {
+pub struct Id<T> {
     id: H256,
     #[serde(skip)]
     _shadow: std::marker::PhantomData<fn() -> T>,
 }
 
+impl<T: Debug + TypeName> Debug for Id<T> {
```

Comment:
I believe the `Debug` bound could be dropped

Reply:
Done